stripes-spring/src/main/java/org/stripesframework/spring/SpringHelper.java

```diff
@@ -93,12 +95,22 @@ public static Object findSpringBean( ApplicationContext ctx, String name, Class<
          }
       } else if ( beanNames.length > 1 ) {
          boolean found = false;
+         final AutowireCapableBeanFactory autowireCapableBeanFactory = ctx.getAutowireCapableBeanFactory();
+         BeanDefinitionRegistry beanDefinitionRegistry = null;
+         if ( autowireCapableBeanFactory instanceof BeanDefinitionRegistry ) {
+            beanDefinitionRegistry = (BeanDefinitionRegistry)autowireCapableBeanFactory;
+         }
          for ( String beanName : beanNames ) {
             if ( beanName.equals(name) ) {
                found = true;
                break;
+            } else if ( beanDefinitionRegistry != null && beanDefinitionRegistry.getBeanDefinition(beanName).isPrimary() ) {
```

Comment:
Could `beanDefinitionRegistry.getBeanDefinition(beanName)` in any case return `null`? Probably not, since we get them from Spring as registered beans before, right? I think that should be fine.

Do you know how beanName and primary is prioritized in Spring? Currently it is a bit random since we take the first in the loop that matches. Maybe we should adjust this to two loops, in the order depending how Spring is doing this lookup?

Reply:
Hi, 
`beanDefinitionRegistry.getBeanDefinition(beanName)` cannot return null.
And this is how spring does it:

		`String[] candidateNames = getBeanNamesForType(requiredType);

		if (candidateNames.length > 1) {
			List<String> autowireCandidates = new ArrayList<>(candidateNames.length);
			for (String beanName : candidateNames) {
				if (!containsBeanDefinition(beanName) || getBeanDefinition(beanName).isAutowireCandidate()) {
					autowireCandidates.add(beanName);
				}
			}
			if (!autowireCandidates.isEmpty()) {
				candidateNames = StringUtils.toStringArray(autowireCandidates);
			}
		}`´

I'll adjust my code accordingly, then I think It's fine.

Thank you